### PR TITLE
samples: lwm2m_client: update overlays

### PR DESCRIPTION
**Pull request by VeijoPesonen**
* Improve the overlay documentation
* Adds low-power-overlay which disables serial port

### CODE REVIEW

**Comment by VeijoPesonen** [comment]
@umapraseeda, @jheiskan81 Please review.

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/sample_description.rst
Comment (VeijoPesonen) [reply samples/nrf9160/lwm2m_client/sample_description.rst:529]: The overlay makes it possible to run the tests but it won't start execution.

**Comment by VeijoPesonen** [comment]
@umapraseeda I applied all your suggestions.

**Review by umapraseeda — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@umapraseeda I assume _doc-required_-label can be removed now. Same goes for the _changelog-entry-required_-label

**Comment by umapraseeda** [comment]
> @umapraseeda I assume _doc-required_-label can be removed now. Same goes for the _changelog-entry-required_-label

Doc required label is automatically added if the PR edits RSTs. That should stay and the PR can be merged if you have a techwriter approval. Removing changelog-required label as no entry is required in the changelog (discussed and agreed offline)

**Pull request merged**
